drivers/accel/adxl314/adxl314.c

```diff
@@ -0,0 +1,1146 @@
+			raw_z[idx] = ((int16_t)xyz_values[buff_idx+5] << 8) + (xyz_values[buff_idx+4]);
+		}
+	}
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Read converted values from FIFO.
+ *
+ * @param dev       - The device structure.
+ * @param entries   - Number of read entries.
+ * @param x  	    - X axis fractional data buffer.
+ * @param y  	    - Y axis fractional data buffer.
+ * @param z  	    - Z axis fractional data buffer.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int adxl314_get_fifo_data(struct adxl314_dev *dev,
+			  uint8_t *entries,
+			  struct adxl314_frac_repr *x,
+			  struct adxl314_frac_repr *y,
+			  struct adxl314_frac_repr *z)
+{
+	int ret = -EINVAL;
+	int16_t raw_x[ADXL314_MAX_FIFO_ENTRIES] = {0};
+	int16_t raw_y[ADXL314_MAX_FIFO_ENTRIES] = {0};
+	int16_t raw_z[ADXL314_MAX_FIFO_ENTRIES] = {0};
+
+	ret = adxl314_get_raw_fifo_data(dev, entries, raw_x, raw_y, raw_z);
+	if (ret)
+		return ret;
+
+	if ((*entries) > 0) {
+		for (uint8_t idx = 0; idx < (*entries); idx++) {
+			x[idx].integer = no_os_div_s64_rem(adxl314_accel_conv(raw_x[idx]),
+							   ADXL314_ACC_SCALE_FACTOR_DIV, &(x[idx].fractional));
+			y[idx].integer = no_os_div_s64_rem(adxl314_accel_conv(raw_y[idx]),
+							   ADXL314_ACC_SCALE_FACTOR_DIV, &(y[idx].fractional));
+			z[idx].integer = no_os_div_s64_rem(adxl314_accel_conv(raw_z[idx]),
+							   ADXL314_ACC_SCALE_FACTOR_DIV, &(z[idx].fractional));
+		}
+	}
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Configure the activity threshold register.
+ *        The scale factor is 780 mg/LSB.
+ *
+ * @param dev     - The device structure.
+ * @param act_thr - Activity threshold value.
+ *
+ * @return ret    - Result of the configuration action.
+ *                  0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int adxl314_conf_act_thr(struct adxl314_dev *dev, uint8_t act_thr)
+{
+	int ret = -EINVAL;
+	uint8_t data[1] = {act_thr};
+
+	ret = adxl314_write_device_data(dev, ADXL314_REG_THRESH_ACT, 1, data);
+
+	if (!ret)
+		dev->act_thr = act_thr;
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Configure the inactivity threshold register.
+ *        The scale factor is 780 mg/LSB.
+ *
+ * @param dev       - The device structure.
+ * @param inact_thr - Inactivity threshold value.
+ *
+ * @return ret      - Result of the configuration action.
+ *                    0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int adxl314_conf_inact_thr(struct adxl314_dev *dev, uint8_t inact_thr)
+{
+	int ret = -EINVAL;
```

Comment:
same as above no initialization needed, check `ret` after `write_device_data` and return 0 at the end of the function. applies for most of the functions below.